src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputParenPad.java

```diff
@@ -131,4 +131,28 @@ public boolean bar(boolean a) {
         int i = (int) (2 * (4 / 2)
                 ); 
     };
+
+    public void myMethod() {
+        String s = "test";
+        Object o = s;
+        ((String)o).length();
+        ( (String)o ).length();
+    }
+
+    public void crisRon() {
+        Object leo = "messi";
+        Object ibra = leo;
+        ((String)leo).compareTo( (String)ibra );
```

Comment:
please add a bit more complicated test cases in inputs, you can take ideas from your report.
combined cases and :
`Boolean value = (Boolean) ( (TwoWayFieldBridge) fieldBridge ).get( field.name(), document );`
`Query query = ( (QueryWrapperFilter) luceneFilter ).getQuery();`
`assertThat( ( (Number) projection[0] ).intValue() ).describedAs( "fieldNotInMapping" ).isEqualTo( 42 );`
.... please add about 10 different and complicated cases.